File: algorithms/455.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int findContentChildren(vector<int> &g, vector<int> &s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        int result = 0, i, j;
        for (i = 0, j = 0; i < g.size(); i++, j++) {
            while (j < s.size() && g[i] > s[j])
                j += 1;
            if (j < s.size())
                result += 1;
        }
        return result;
    }
};

int main() {
    return 0;
}

File: algorithms/rust/455.rs
impl Solution {
    #[allow(dead_code)]
    pub fn find_content_children(mut g: Vec<i32>, mut s: Vec<i32>) -> i32 {
        g.sort();
        s.sort();
        let mut rslt = 0;
        let mut g_iter = g.iter().rev();
        for cookie in s.iter().rev() {
            for child in &mut g_iter {
                if cookie >= child {
                    rslt += 1;
                    break;
                }
            }
        }
        rslt
    }
}

struct Solution;

fn main() {
    println!("Hello, world!");
}
